From 411d20f8bcec402c61d3f7b4ba1877c29c66c84e Mon Sep 17 00:00:00 2001
From: Sadaf Ebrahimi <sadafebrahimi@google.com>
Date: Mon, 23 May 2022 22:34:43 +0000
Subject: [PATCH] Prevent integer overflow in copyString

Bug: http://b/221384482
Change-Id: Ibdcb5dc24ee8886a04c2e29bd6ddccf29ece73ad
(cherry picked from commit 72de6c81aa02d48ed86b96d2e29d773086fb7d4c)
Merged-In: Ibdcb5dc24ee8886a04c2e29bd6ddccf29ece73ad
---
 lib/xmlparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index 3aaf35b..5c52e05 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -6875,7 +6875,7 @@ getElementType(XML_Parser parser, const ENCODING *enc, const char *ptr,
 
 static XML_Char *
 copyString(const XML_Char *s, const XML_Memory_Handling_Suite *memsuite) {
-  int charsRequired = 0;
+  size_t charsRequired = 0;
   XML_Char *result;
 
   /* First determine how long the string is */
-- 
2.37.1.359.gd136c6c3e2-goog

